src/coverlet.msbuild.tasks/InstrumentationTask.cs

```diff
@@ -119,6 +121,16 @@ public override bool Execute()
         {
             WaitForDebuggerIfEnabled();
 
+            IServiceCollection serviceCollection = new ServiceCollection();
+            serviceCollection.AddSingleton<IRetryHelper, RetryHelper>();
+            serviceCollection.AddTransient<IProcessExitHandler, ProcessExitHandler>();
+            serviceCollection.AddTransient<IFileSystem, FileSystem>();
+            serviceCollection.AddTransient<IConsole, SystemConsole>();
+            serviceCollection.AddTransient<ILogger, MSBuildLogger>(x => _logger);
+            serviceCollection.AddSingleton<IInstrumentationHelper, InstrumentationHelper>();
```

Comment:
Add a comment `// We need to keep singleton/static semantics` like in old DI class

Reply:
Done